src/Illuminate/Database/Eloquent/Model.php

```diff
@@ -2748,6 +2748,23 @@ protected function hasCast($key)
     }
 
     /**
+     * Determine whether a value is Date / DateTime castable for inbound manipulation.
+     *
+     * @param  string  $key
+     * @return bool
+     */
+    protected function isDateCastable($key)
+    {
+        if ($this->hasCast($key)) {
+            return in_array(
+                $this->getCastType($key), ['date', 'datetime'], true
```

Comment:
was just following the format that isJsonCastable() set below it... should this change in both?